lib/Zonemaster/CLI.pm

```diff
@@ -631,31 +663,24 @@ sub run {
     # Actually run tests!
     eval {
         if ( $self->test and @{ $self->test } > 0 ) {
-            my $zone = Zonemaster::Engine->zone( $domain );
-            foreach my $t ( @{ $self->test } ) {
-                # The case does not matter
-                $t = lc( $t );
-                # Fully qualified module and test case (e.g. Example/example12)
-                if (my ($module, $method) = $t =~ m#^ ( [a-z]+ ) / ( [a-z]+[0-9]{2} ) $#ix) {
-                    Zonemaster::Engine->test_method( $module, $method, $zone );
-                }
-                # Just a test case (e.g. example12). Note the different capturing order.
-                elsif (($method, $module) = $t =~ m#^ ( ( [a-z]+ ) [0-9]{2} ) $#ix) {
-                    Zonemaster::Engine->test_method( $module, $method, $zone );
+            foreach my $t ( @testing_suite ) {
+                # Either a module/method, or just a module
+                my ( $module, $method ) = split('/', $t);
+                if ( $method ) {
+                    # Force Engine to include the requested test case in its profile
+                    Zonemaster::Engine::Profile->effective->set( 'test_cases', [ "$method" ] );
```

Comment:
The test_cases property is set here but its original value is never reset. I believe this messes with the call to test_module() below. E.g. when running `zonemaster-cli --test connectivity01 --test consistency example.com`.